keadm/cmd/keadm/app/cmd/edge/join_others.go

```diff
@@ -237,6 +238,19 @@ func join(opt *common.JoinOptions, step *common.Step) error {
 }
 
 func runEdgeCore(withMqtt bool) error {
+	//If selinux is enabled, it is necessary to modify the context of edgecore, as edgecore is copied from the container
```

Comment:
The logic here is simple, and most of it is for error return judgment. I don't think it's necessary to add an additional function.

Reply:
> I think this code is too deeply nested; perhaps it can be modified to:
> 
> ```
> func runEdgeCore(withMqtt bool) error{
> 	err := selinuxLabelRevision(selinux.GetEnabled())
> 	if err != nil {
> 		return err
> 	}
> ...
> }
> 
> func selinuxLabelRevision(enable bool) error {
> 	if !enable {
> 		return nil
> 	}
> 
> 	label, err := selinux.FileLabel(filepath.Join(util.KubeEdgeUsrBinPath, util.KubeEdgeBinaryName))
> 	if err != nil {
> 		return fmt.Errorf("get selinux context of edgecore faild with error:%w", err)
> 	}
> 
> 	if label == util.EdgeCoreSELinuxLabel {
> 		return nil
> 	}
> 
> 	if err = selinux.SetFileLabel(filepath.Join(util.KubeEdgeUsrBinPath, util.KubeEdgeBinaryName), util.EdgeCoreSELinuxLabel); err != nil {
> 		return fmt.Errorf("reset selinux context on edgecore faild with error:%w", err)
> 	}
> 	return nil
> }
> ```

ok, I have made another adjustment based on the suggestions you provided.